tests/integration/conftest.py

```diff
@@ -34,34 +36,35 @@ async def charm_fixture(request: pytest.FixtureRequest, ops_test: OpsTest) -> st
     return charm
 
 
-@pytest.fixture(scope="module", name="model")
+@pytest.fixture(scope="function", name="model")
 def model_fixture(ops_test: OpsTest) -> Model:
     """The testing model."""
     assert ops_test.model
     return ops_test.model
 
 
-@pytest.fixture(scope="module", name="num_agents")
-def num_agents_fixture() -> int:
-    """The number of agents to deploy."""
-    return 1
-
-
-@pytest_asyncio.fixture(scope="module", name="jenkins_agent_application")
+@pytest_asyncio.fixture(
+    scope="function", name="jenkins_agent_application", params=["focal", "jammy"]
+)
 async def application_fixture(
-    model: Model, charm: str, num_agents: int
+    model: Model, charm: str, request: typing.Any
```

Comment:
Is there any replacement for this Any?

Reply:
From what I've found, unfortunately there are nothing out-of-the box that we can use to type hint this `request` parameter: https://stackoverflow.com/questions/65334215/how-can-request-param-be-annotated-in-indirect-parametrization